.mvn/wrapper/maven-wrapper.properties

```diff
@@ -0,0 +1,18 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+distributionUrl=https://repo.maven.apache.org/maven2/org/apache/maven/apache-maven/3.8.1/apache-maven-3.8.1-bin.zip
+wrapperUrl=https://repo.maven.apache.org/maven2/org/apache/maven/wrapper/maven-wrapper/3.2.0/maven-wrapper-3.2.0.jar
```

Comment:
That's a good point 👍🏻 
Does Maven provide an option to generate/get the `wrapperSha256Sum` and `distributionSha256Sum`?

Reply:
@ggrossetie I looked but couldn't find anything, I believe one has to manually set these when updating the `maven-wrapper.properties`, making it effectively trust-on-first-use.

I do notice they PGP-sign the binaries as well, the `.asc` file is available from https://repo.maven.apache.org/maven2/org/apache/maven/wrapper/maven-wrapper/3.2.0/, but how to use this properly seems not-so-easy to implement compared to pinning checksums.

In general I also tried looking into dependency checksum verification for the `pom.xml`, but this appears to be a real rabbit hole with Maven. Ideally all dependencies that are pinned have their checksum pinned in the Kroki repository too to prevent man-in-the-middle supply chain attacks. `package-lock.json` does this for node, but I couldn't really find info about this with Maven and only some vague hits with Gradle.

Personally not familiar with Java ecosystem at all, this might be worth a follow-up issue though. Your thoughts on this?

Reply:
> Personally not familiar with Java ecosystem at all, this might be worth a follow-up issue though. Your thoughts on this?

Yes please, open a new issue so we can discuss how to address this concern 👍🏻

Reply:
@ggrossetie Done, ended up a bit broader/bigger than expected but hopefully will be of good use! :p

See #1551